cmake/Version.cmake

```diff
@@ -6,22 +6,62 @@
 #
 #
 
+include(cmake/FileUtil.cmake)
+
 string(TIMESTAMP current_year "%Y")
 set(O3DE_COPYRIGHT_YEAR ${current_year} CACHE STRING "Open 3D Engine's copyright year")
-set(O3DE_VERSION_STRING "0.0.0" CACHE STRING "Open 3D Engine's version")
-set(O3DE_DISPLAY_VERSION_STRING "00.00" CACHE STRING "Open 3D Engine's display version")
-set(O3DE_BUILD_VERSION 0 CACHE STRING "Open 3D Engine's build number")
-set(O3DE_ENGINE_NAME "o3de" CACHE STRING "Open 3D Engine's engine name")
-
-# Optional environment overrides
-if(NOT "$ENV{O3DE_VERSION}" STREQUAL "")
-    set(O3DE_VERSION_STRING "$ENV{O3DE_VERSION}")
-endif()
-
-if(NOT "$ENV{O3DE_DISPLAY_VERSION}" STREQUAL "")
-    set(O3DE_DISPLAY_VERSION_STRING "$ENV{O3DE_DISPLAY_VERSION}")
-endif()
-
-if(NOT "$ENV{O3DE_BUILD_VERSION}" STREQUAL "")
-    set(O3DE_BUILD_VERSION "$ENV{O3DE_BUILD_VERSION}")
-endif()
+
+get_filename_component(ENGINE_JSON_PATH "${CMAKE_CURRENT_LIST_DIR}/../engine.json" ABSOLUTE)
+ly_file_read(${ENGINE_JSON_PATH} manifest_json_data)
+
+#! o3de_read_engine_default: Attempts to read a field from engine.json 
+#     that might be overridden by an environment variable. Priority order:
+#      1. environment variable
+#      2. engine.json value
+#      3. default value
+#  
+# \arg:output_value - name of output variable to set 
+# \arg:key - name of field in engine.json 
+# \arg:env_value - name of environment variable to use as override 
+# \arg:default_value - value to use if neither environment var or engine.json value found 
+function(o3de_read_engine_default output_value env_value key default_value description)
+    # use Environment override if provided
+    if(NOT ${env_value} STREQUAL "")
+        set(${output_value} ${env_value} CACHE STRING ${description})
+        return()
+    endif()
```

Comment:
Checking that the ENV variable isn't an empty string is fine for the version fields later on in this file.
But if there is ever a time when the environment variable can exist, but be empty, then it would be better to use the cmake [if(DEFINED ENV{var})](https://cmake.org/cmake/help/latest/command/if.html?highlight=#existence-checks) check